clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -934,6 +935,44 @@ mlir::SuccessorOperands BrOp::getSuccessorOperands(unsigned index) {
 
 Block *BrOp::getSuccessorForOperands(ArrayRef<Attribute>) { return getDest(); }
 
+static void eraseBlock(Block *block) {
+  for (auto &op : llvm::make_early_inc_range(*block))
+    op.erase();
+  block->erase();
+  return;
+}
+
+/// Removes branches between two blocks if it is the only branch.
+///
+/// From:
+///   ^bb0:
+///     cir.br ^bb1
+///   ^bb1:  // pred: ^bb0
+///     cir.return
+///
+/// To:
+///   ^bb0:
+///     cir.return
+LogicalResult BrOp::fold(FoldAdaptor adaptor,
```

Comment:
There's both `mlir::LLVM::BrOp` and `mlir::cir::BrOp`, if this is the reason why you are adding LLVM stuff here please use the complete namespace.